api/v1beta1/clusterclass_types.go

```diff
@@ -583,6 +583,54 @@ type JSONSchemaProps struct {
 	// It can be used to add additional data for higher level tools.
 	// +optional
 	XMetadata *VariableSchemaMetadata `json:"x-metadata,omitempty"`
+
+	// x-kubernetes-int-or-string specifies that this value is
+	// either an integer or a string. If this is true, an empty
+	// type is allowed and type as child of anyOf is permitted
+	// if following one of the following patterns:
+	//
+	// 1) anyOf:
+	//    - type: integer
+	//    - type: string
+	// 2) allOf:
+	//    - anyOf:
+	//      - type: integer
+	//      - type: string
+	//    - ... zero or more
+	// +optional
+	XIntOrString bool `json:"x-kubernetes-int-or-string,omitempty"`
```

Comment:
doesn't this field break the naming convention of camelCase?
why not `xIntOrString` ?

Reply:
Our primary goal here is to keep this as consistent as possible with the Kubernetes CRD API type.

So basically that folks who are familiar with Kubernetes CRDs can write ~ the same schema with variables (without having to map keywords)

(That being said, I don't know the reasons why Kubernetes itself chose to have the markers in this format)

Reply:
But seems like they are sticking to it even for newer features like CEL, xref: https://github.com/kubernetes/kubernetes/blob/67a171a1422cc5861491aadd69e51ce718196434/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go#L196

Reply:
OpenAPI extensions are custom properties prefixed by `x-` and it seems it is more common to use kebab-case for these extensions in the wild than camelCase for these extensions as opposed to a mix like `x-intOrString`. I assume the `kubernetes` part is there to show this is a kubernetes specific openapi extension. More details at https://swagger.io/docs/specification/v3_0/openapi-extensions/.